packages/react-app/src/pages/location.tsx

```diff
@@ -0,0 +1,248 @@
+  const navigate = useNavigate();
+
+  const { handleSubmit, register, formState, setValue, control } = useForm<Location>({});
+
+  const [position, setPosition] = useState(defaultPin);
+  const [mapRadius, setRadius] = useState(defaultRadius);
+
+  function DraggableMarker() {
+    const markerRef = useRef<any>(null)
+    const eventHandlers = useMemo(
+      () => ({
+        dragend() {
+          const marker = markerRef.current
+          if (marker) {
+            setPosition(marker.getLatLng())
+          }
+        },
+      }),
+      [],
+    )
+
+    return (
+      <>
+        <Circle
+          color='transparent'
+          center={position}
+          eventHandlers={eventHandlers}
+          pathOptions={{ fillColor: '#12BB6A' }}
+          radius={mapRadius}>
+        </Circle>
+        <Marker
+          draggable={true}
+          eventHandlers={eventHandlers}
+          position={position}
+          ref={markerRef}>
+        </Marker>
+      </>
+    )
+  }
+
+  const onSave = (data: Location) => {
+    FrontendApiClient.postLocation(data)
+      .then(() => {
+        navigate(-1);
+      })
+      .catch(e => {
+        setErrorMessage(e.message);
+      });
+  }
+
+  const [errorMessage, setErrorMessage] = useState<string | undefined>(undefined);
+
+  return (
+    <Box>
+      <Snackbar
+        open={!!errorMessage}
+        autoHideDuration={6000}
+        onClose={() => setErrorMessage(undefined)}
+        anchorOrigin={{ vertical: 'bottom', horizontal: 'right' }}
+      >
+        <MuiAlert severity="error" onClose={() => setErrorMessage(undefined)}>
+          {errorMessage}
+        </MuiAlert>
+      </Snackbar>
+      <SubHeader />
+      <Typography
+        sx={{
+          fontSize: '20px',
+          fontWeight: 'bold',
+          color: '#303136',
+          marginTop: '13px',
+          marginLeft: { xs: '10px', md: '0px' },
+          px: '5rem',
+        }}
+      >
+        New location
+      </Typography>
+      <Box sx={{ px: '5rem', pt: '1rem' }}>
+        <Grid container spacing={5}>
+          <Grid item xs={8}>
+            <MapContainer center={position} zoom={16} scrollWheelZoom={false} style={{ height: '100%', minHeight: 500 }}>
+              <TileLayer
+                attribution='&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
+                url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
+              />
+              <DraggableMarker />
+            </MapContainer>
+          </Grid>
+          <Grid item xs={4}>
+            <Box sx={{ mb: '1.5rem' }}>
+              <Typography>
+                Name
+              </Typography>
+              <TextField
+                id="name"
+                variant="outlined"
+                fullWidth
+                size="small"
+                sx={textInputStyle}
+                {...register('name', { required: 'Required' })}
+                error={formState.errors.name ? true : false}
```

Comment:
Just a shorter way to do the same thing
```suggestion
                error={!!formState.errors.name}
```